evennia/web/website/forms.py

```diff
@@ -0,0 +1,142 @@
+        """
+        Randomly distributes a number of points across the given attributes,
+        while also ensuring each attribute gets at least a certain amount
+        and at most a certain amount.
+        
+        Args:
+            attribute_list (iterable): List or tuple of attribute names to assign
+                points to.
+            points (int): Starting number of points
+            min_points (int): Least amount of points each attribute should have
+            max_points (int): Most amount of points each attribute should have
+            
+        Returns:
+            spread (dict): Dict of attributes and a point assignment.
+            
+        """
+        num_buckets = len(attribute_list)
+        point_spread = (x for x in self.random_distribution(points, num_buckets, min_points, max_points))
+        
+        # For each field, get the point calculation for the next attribute value generated
+        return {attribute: next(point_spread) for k in attribute_list}
+    
+    @classmethod
+    def random_distribution(cls, points, num_buckets, min_points, max_points):
+        """
+        Distributes a set number of points randomly across a number of 'buckets'
+        while also attempting to ensure each bucket's value finishes within a 
+        certain range.
+        
+        If your math doesn't add up (you try to distribute 5 points across 100
+        buckets and insist each bucket has at least 20 points), the algorithm
+        will return the best spread it could achieve but will not raise an error
+        (so in this case, 5 random buckets would get 1 point each and that's all).
+        
+        Args:
+            points (int): The number of points to distribute.
+            num_buckets (int): The number of 'buckets' (or stats, skills, etc)
+                you wish to distribute points to.
+            min_points (int): The least amount of points each bucket should have.
+            max_points (int): The most points each bucket should have.
+            
+        Returns:
+            buckets (list): List of random point assignments.
+        
+        """
+        buckets = [0 for x in range(num_buckets)]
+        indices = [i for (i, value) in enumerate(buckets)]
+    
+        # Do this while we have eligible buckets, points to assign and we haven't
+        # maxed out all the buckets.
+        while indices and points and sum(buckets) <= (max_points * num_buckets):
+            # Pick a random bucket index
+            index = choice(indices)
+           
+            # Add to bucket
+            buckets[index] = buckets[index] + 1
+            points = points - 1
+            
+            # Get the indices of eligible buckets
+            indices = [i for (i, value) in enumerate(buckets) if (value < min_points) or (value < max_points)]
+            
+        return buckets
+    
+class ExtendedCharacterCreationForm(forms.Form):
```

Comment:
This definitely does not belong in core. This should rather be a contrib or - maybe better - a wiki tutorial.